### PR TITLE
Add health check for object detection pipeline

### PR DESCRIPTION
**Pull request by vangork**
add probe before buffer goes into pravegasink.

Test:
```
GST_PLUGIN_PATH=./target/debug:${GST_PLUGIN_PATH} PROTO_LIB=./target/debug/libnvds_pravega_proto.so ALLOW_CREATE_SCOPE=false GST_DEBUG="WARNING" INPUT_STREAM=examples/camera-claudio-06 MSGAPI_CONFIG_FILE=$HOME/msgapi-config.txt LOG_LEVEL=10 OUTPUT_VIDEO_STREAM=examples/osd-luis-01 pravega_client_tls_cert_path=/etc/ssl/certs/ca-certificates.crt PRAVEGA_CONTROLLER_URI="tls://pravega-controller.kubespray.nautilus-platform-dev.com:443" KEYCLOAK_SERVICE_ACCOUNT_FILE=$HOME/examples.json RUST_LOG="off" HEALTH_CHECK_ENABLED=true  HEALTH_CHECK_IDLE_SECONDS=30 ./deepstream/python_apps/deepstream-pravega-demos/pravega-to-object-detection-to-pravega.py
```

### CODE REVIEW

File: deepstream/python_apps/deepstream-pravega-demos/pravega-to-object-detection-to-pravega.py [hunk 1]
```diff
@@ -636,6 +709,11 @@ def main():
             # Always write to Pravega immediately regardless of PTS
             pravegasink.set_property("sync", False)
             pravegasink.set_property("timestamp-mode", "tai")
+            if args.health_check_enabled:
+                idle_detector = IdleDetector(args.health_check_idle_seconds)
+                HealthCheckHttpHandler.idle_detector = idle_detector
+                pravegasinkpad = pravegasink.get_static_pad("sink")
```
Comment (claudiofahey) [inline deepstream/python_apps/deepstream-pravega-demos/pravega-to-object-detection-to-pravega.py:715]: You also need to add this probe to the msgbroker element.

**Review by claudiofahey — CHANGES_REQUESTED** [review]

File: deepstream/python_apps/deepstream-pravega-demos/pravega-to-object-detection-to-pravega.py (hunk 1 above)
Comment (vangork) [reply deepstream/python_apps/deepstream-pravega-demos/pravega-to-object-detection-to-pravega.py:715]: But we can't gurantee object detected in a specific period.

Comment (claudiofahey) [reply deepstream/python_apps/deepstream-pravega-demos/pravega-to-object-detection-to-pravega.py:715]: Ok I see. In that case, put the probe on pravegasrc or some other common element. If anything goes wrong with the pipeline, pravegasrc will eventually block when the queues fill up.

**Review by claudiofahey — APPROVED** [review]
I moved all common code to a new Python library and made a few other changes. Please see my commits.

**Pull request merged**